client/src/components/related/RelatedProducts.jsx

```diff
@@ -48,128 +47,36 @@ class RelatedProducts extends React.Component {
-
-    if (toLoad.length) {
-      toLoad.forEach(id => {
-        axios.get(`/products/${id}`)
-          .then(res => {
-            const product = res.data;
-            const relatedProducts = [ ...this.state.related ];
-            // console.log(`Related product ${product.id} (${product.name}) retrieved from server`);
-            updateCache('products', id, product);
-            this.setState({
-              related: [ ...relatedProducts, product ]
-            });
-          })
-          .catch(err => {
-            console.log(err.stack);
-          });
-      });
-    }
-
-    ///////////////////////////////////////////////////////////////////////////////////////////
-    // Something like the following could be used if we decide to limit client HTTP requests //
-    ///////////////////////////////////////////////////////////////////////////////////////////
-    //   return axios.get(`/multipleProducts?ids=${load.join('&ids=')}`)
-    //     .then(res => {
-    //       console.log(res.data);
-    //       if (loaded.length) {
-    //         loaded.concat(res.data);
-    //       } else {
-    //         loaded = res.data;
-    //       }
-    //     })
-    //     .catch(err => {
-    //       console.log(err.stack);
-    //     });
-    // } else {
-    //   if (loaded.length) {
-    //     return Promise.resolve(loaded);
-    //   } else {
-    //     return Promise.reject('No product data');
-    //   }
-    // }
-  }
-
-  loadOutfit() {
-    const { localStorage } = window;
-    const outfitData = localStorage.getItem('outfit');
-
-    if (outfitData) {
-      const outfit = JSON.parse(outfitData);
-      // console.log('Outfit found in localStorage:', outfit);
-      const { checkCache, updateCache } = this.props;
-      let cached = [];
-      let index = 0;
-
-      outfit.forEach(id => {
-        let product = checkCache('products', id);
-
-        if (product) {
-          cached[index++] = product;
-          // console.log(`Outfit product ${product.id} (${product.name}) loaded from cache`);
-          this.setState({ outfit: [ ...cached ] });
-        } else {
-          const asyncIndex = index++;
-          axios.get(`/products/${id}`)
-            .then(res => {
-              const product = res.data;
-              cached[asyncIndex] = product;
-              // console.log(`Outfit product ${product.id} (${product.name}) retrived from server`);
-              this.setState({ outfit: [ ...cached ] });
-              updateCache('products', product.id, product);
-            })
-            .catch(err => {
-              console.log('Loading outfit:', err.stack);
-            });
-        }
-      });
-    }
-  }
-
-  updateOutfit(newOutfit) {
-    this.setState({ outfit: newOutfit });
-  }
+  // updateOutfit(newOutfit) {
+  //   this.setState({ outfit: newOutfit });
+  // }
 
   selectProduct(product) {
     this.props.selectProduct(product);
-    this.setState({ ready: false });
+    this.update = true;
```

Comment:
Just a casual observation, but perhaps more robust code would have the select callback return a boolean indicating success/failure, and then you can make something like 'this.update' contingent upon the callback succeeding and returning 'true'?

Reply:
Good call. This was part of some testing I was doing. It's dead code.